compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUConfigureTensorLayouts.cpp

```diff
@@ -48,6 +48,35 @@ static SmallVector<bool> getPromotedOperands(Operation *op) {
   return promotedOperands;
 }
 
+static IREE::GPU::MmaInterfaceAttr getIntrinsic(Operation *op) {
+  auto config = getLoweringConfig<IREE::GPU::LoweringConfigAttr>(op);
+  assert(config && "Cannot find intrinsic from unconfigured op.");
+
+  IREE::GPU::MmaInterfaceAttr mmaIntrinsic = config.getMmaKind();
+  assert(mmaIntrinsic && "Cannot find intrinsic in lowering config.");
+  return mmaIntrinsic;
+}
+
+static int64_t getSubgroupMCount(Operation *op) {
+  auto config = getLoweringConfig<IREE::GPU::LoweringConfigAttr>(op);
+  assert(config && "Cannot find intrinsic from unconfigured op.");
+
+  std::optional<int64_t> subgroup_m_count = config.getSubgroupMCount();
+  assert(subgroup_m_count.has_value() &&
+         "Cannot find subgroup_m_count in lowering config.");
+  return subgroup_m_count.value();
```

Comment:
IIRC LLVM is configured to assert on empty optional dereference by default. I think you can turn this and the below into `return *config.getSubgroupMCount();`

Reply:
Didn't know that, that's really nice! done!